src/bundler.ts

```diff
@@ -488,64 +499,115 @@ class Bundler {
       for (const bundleEntry of manifest.bundles) {
         const bundleUrl = bundleEntry[0];
         const bundle = {url: bundleUrl, bundle: bundleEntry[1]};
-        const bundledAst = await this._bundleDocument(
-            bundleUrl, bundle, manifest, bundle.bundle.files);
-        bundledDocuments.set(bundleUrl, bundledAst);
+        const bundledAst =
+            await this._bundleDocument(bundle, manifest, bundle.bundle.files);
+        bundledDocuments.set(
+            bundleUrl,
+            {ast: bundledAst, files: Array.from(bundle.bundle.files)});
       }
       return bundledDocuments;
     }
   }
 
-  private async _bundleDocument(
-      url: string,
-      bundle: AssignedBundle,
-      bundleManifest: BundleManifest,
-      bundleImports?: Set<string>): Promise<ASTNode> {
-    const analyzedRoot = await this.analyzer.analyze(url);
-
-    // Set tracking imports that have been reached.
-    const inlinedImports: Set<UrlString> = new Set();
-
-    // We must parse document to a new AST since we will be modifying the AST.
-    const newDocument = parse5.parse(analyzedRoot.parsedDocument.contents);
+  private _computeRelativeUrl(from: UrlString, to: UrlString): string
```

Comment:
We already have this:
https://github.com/Polymer/vulcanize/blob/v3/src/url-utils.ts#L44

Which has lots of tests because tricky cases:
https://github.com/Polymer/vulcanize/blob/v3/src/test/url-utils_test.ts#L67